File: 14.最长公共前缀.cs
/*
 * @lc app=leetcode.cn id=14 lang=csharp
 *
 * [14] 最长公共前缀
 */

using System;

// @lc code=start
class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        if (strs.Length == 1)
            return strs[0];
        if (strs[0].Length == 0)
            return strs[0];

        for (short i = 0; true; i++)
        {
            if (strs[0].Length == i)
            {
                return strs[0].Substring(0, i);
            }

            for (short j = 1; j < strs.Length; j++)
            {
                if (strs[j].Length == i || strs[j][i] != strs[0][i])
                {
                    return strs[0].Substring(0, i);
                }
            }
        }
    }
}
// @lc code=end

File: 14.最长公共前缀.cpp
/*
 * @lc app=leetcode.cn id=14 lang=cpp
 *
 * [14] 最长公共前缀
 */

#include <vector>
#include <string>

using namespace std;

// @lc code=start
class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        if (strsSize == 1)
            return strs[0];
        if (strs[0].empty())
            return strs[0];

        char *result = strs[0];
        for (uint8_t i = 0; true; i++)
        {
            for (uint8_t j = 1; j < strsSize; j++)
            {
                if (strs[j][i] == '\0' || strs[j][i] != result[i])
                {
                    result[i] = '\0';
                    return result;
                }
            }
            i++;
        }
    }
};
// @lc code=end
